### PR TITLE
Add tooltips for library and feature metadata

### PR DESCRIPTION
**Pull request by wlach**
* Add tooltips for library and feature metadata and use them
  consistently throughout he application
* Add a "page header" component to be used on all pages with tags
  beneath them
* Remove "warning" banners for libraries on metric and ping pages in
  favour of using tags (with these descriptions)
* Fix some bugs in our usage of tippy.js which made descriptions
  not work with the itemlist

### CODE REVIEW

**Comment by Iinh** [comment]
I'm testing with the deploy preview and at first glance it seems that AppDetail page for apps other than fenix is broken. Examples: 
- https://deploy-preview-828--glean-dictionary-dev.netlify.app/apps/bergamot
- https://deploy-preview-828--glean-dictionary-dev.netlify.app/apps/burnham

With the below error:
CleanShot 2021-09-14 at 15 58 40@2x

**Comment by wlach** [comment]
Argh! Thanks @Iinh. More ammunition to spend some time on #697

File: scripts/build-glean-metadata.py [hunk 1]
```diff
@@ -227,6 +241,9 @@ def get_app_variant_description(app):
     if all([app_id.get("prototype") for app_id in app_group["app_ids"]]):
         app_data["prototype"] = True
 
+    app_annotation = _get_annotation(annotations_index, app_name, "app")
+    app_tags = app_annotation.get("tags")
```
Comment (Iinh) [inline scripts/build-glean-metadata.py:245]: Do we need to provide a fall back value for `app_annotation.get("tags")` or does `tags` always exist?

File: src/components/tippy.js [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+  let instance = props && tippy(node, props);
+  return {
+    update(newProps) {
+      if (instance) instance.destroy();
+      if (newProps) instance = tippy(node, newProps);
+    },
+    destroy() {
+      if (instance) instance.destroy();
+    },
+  };
```
Comment (Iinh) [inline src/components/tippy.js:13]: I wasn't able to understand why this change was necessary. According to the PR description, is it to "fix some bugs in our usage of tippy.js which made descriptions not work with the itemlist"?

File: src/pages/AppDetail.svelte [hunk 3]
```diff
@@ -72,7 +74,9 @@
   >
     <Tab key="metrics">Metrics</Tab>
     <Tab key="pings">Pings</Tab>
-    {#if app.tags && app.tags.length}
+
+    {#if false && app.tags && app.tags.length}
+      <!-- commenting this out for now, see: https://github.com/mozilla/glean-dictionary/issues/832 -->
```
Comment (Iinh) [inline src/pages/AppDetail.svelte:79]: 👍

**Review by Iinh — APPROVED** [review]
This looks great to me! I only have some clarifying questions below.

File: scripts/build-glean-metadata.py (hunk 1 above)
Comment (wlach) [reply scripts/build-glean-metadata.py:245]: If the app has no tags then there shouldn't be any annotations with them: it may be best to fallback to `{}` in this case just for clarity though. I'll change this.

File: src/components/tippy.js (hunk 2 above)
Comment (wlach) [reply src/components/tippy.js:13]: In the metric list group, labels may be updated or destroyed as the user paginates through the content. This could result in stale content appearing for a label.

Comment (Iinh) [reply src/components/tippy.js:13]: ah I see what you mean. Thanks for the explanation!
